elementary/exceptions/exceptions.py

```diff
@@ -41,19 +42,30 @@ def __init__(
         self,
         err: subprocess.CalledProcessError,
         base_command_args: List[str],
-        err_msg: Optional[str],
-    ):
+        err_msg: Optional[str] = None,
+        logs: Optional[List[DbtLog]] = None,
+    ):  
         msg = "Failed to run dbt command."
+        if logs and not err_msg:
+            err_msg = '\n'.join([log.msg for log in logs if log.level == 'error'])
         if err_msg:
             msg = f"{msg}\n{err_msg}"
-
         super().__init__(msg)
 
         # Command args sent to _run_command (without additional user-specific args it as such as projects / profiles
         # dir)
         self.proc_err = err
         self.base_command_args = base_command_args
         self.return_code = err.returncode
+        self.logs = logs
+    
+    @classmethod
+    def from_process_error(cls, err: subprocess.CalledProcessError, base_command_args: List[str], err_msg: Optional[str] = None) -> 'DbtCommandError':
```

Comment:
Aren't the `base_command_args` and `err_msg` contained with the `err` object?
Also seems like this isn't used so maybe YAGNI.

By the way, any reason not to `-> DbtCommandError`?

Reply:
`base_command_args` and `err.args` are not the same :(
as for `err_msg`, its a way for the programmer to give a more specific error message, by default it is the error logs